modules/javafx.graphics/src/main/java/javafx/scene/paint/Color.java

```diff
@@ -1935,10 +1940,20 @@ Object acc_getPlatformPaint() {
     }
 
     /**
-     * {@inheritDoc}
+     * Returns an intermediate value between the value of this {@code Color} and the specified
+     * {@code endValue} using the linear interpolation factor {@code t}, ranging from 0 (inclusive)
+     * to 1 (inclusive).
+     *
+     * @param endValue the target value
+     * @param t the interpolation factor
+     * @throws NullPointerException if {@code endValue} is {@code null}
+     * @return the intermediate value
      */
-    @Override public Color interpolate(Color endValue, double t) {
-        if (t <= 0.0) return this;
+    public Color interpolate(Color endValue, double t) {
+        Objects.requireNonNull(endValue, "endValue cannot be null");
+
+        // If both instances are equal, return this instance to prevent the creation of numerous small objects.
+        if (t <= 0.0 || equals(endValue)) return this;
```

Comment:
This question might be outside of the current PR, but I think the naive color interpolation algorithm is a bad choice here (L1960).

See for example
https://www.alanzucconi.com/2016/01/06/colour-interpolation/

The question that is relevant to this PR is whether we should enable the user to specify the interpolation algorithm.

Reply:
CSS Transitions 1 is a feature-complete spec, and it doesn't include an option to specify the transition color space. That might come with CSS Transitions 2, but it's not there yet. I think we should revisit this once the feature has landed in the spec.

Reply:
pardon my ignorance, could you clarify what a "CSS Transition 1" is?

Reply:
It's basically the feature level of the specification.
Here's level 1: https://www.w3.org/TR/css-transitions-1/ (this is implemented in JavaFX)
Here's level 2: https://www.w3.org/TR/css-transitions-2/
There seems to be an ongoing discussion around different color interpolation options, but nothing has made it into the level 2 specification yet.

Reply:
Yes, I think this would be a good enhancement.